lib/core/db_models/worker/education/education.dart

```diff
@@ -27,56 +31,76 @@ class Education {
       this.summary,
       this.workerRef});
 
-  factory Education.fromJson(Map<String, dynamic> json) => Education(
-      educationInstitution: json[EducationSchema.educationInstitution] != null
-          ? EducationInstitution.fromJson(
-              json[EducationSchema.educationInstitution])
-          : null,
-      endDate: json[EducationSchema.endDate] != null
-          ? json[EducationSchema.endDate]
-          : "",
-      startDate: json[EducationSchema.startDate] != null
-          ? json[EducationSchema.startDate]
-          : "",
-      award: json[EducationSchema.award] != null
-          ? json[EducationSchema.award]
-          : false,
-      location: json[EducationSchema.location] != null
-          ? json[EducationSchema.location]
-          : "",
-      course: json[EducationSchema.course] != null
-          ? Course.fromJson(json[EducationSchema.course])
-          : null,
-      level: json[EducationSchema.level] != null
-          ? Level.fromJson(json[EducationSchema.level])
-          : null,
-      grade: json[EducationSchema.grade] != null
-          ? Grade.fromJson(json[EducationSchema.grade])
-          : null,
-      knowledgeAreaList: json[EducationSchema.knowledgeAreas] != null
-          ? List<KnowledgeArea>.from(json[EducationSchema.knowledgeAreas]
-              .map((x) => KnowledgeArea.fromJson(x)))
-          : [],
-      summary: json[EducationSchema.summary] != null
-          ? json[EducationSchema.summary]
-          : "",
-      workerRef: json[EducationSchema.workerRef] != null
-          ? json[EducationSchema.workerRef]
-          : "");
+  bool checkAllFieldsNull() {
+    if (endDate != null ||
+        location != null ||
+        educationInstitution != null ||
+        course != null ||
+        award != null ||
+        startDate != null ||
+        summary != "" ||
+        level != null ||
+        grade != null ||
+        knowledgeAreaList.length > 0) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  factory Education.fromJson(Map<String, dynamic> json, {String docId}) =>
```

Comment:
Since startDate and Time are changed from String to Timestamp, Please refer to WorkHistoryModel to see how timestamps are managed if start and endnote are Strings because on Organisation portal they are coming as string and same model are used